ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -827,29 +826,13 @@ impl<'a> X64CodeGen<'a> {
 
     /// Load a constant into the specified register.
     fn load_const(&mut self, reg: Rq, cidx: jit_ir::ConstIdx) {
-        let cst = self.m.const_(cidx);
-        let mut bytes = cst.bytes().as_slice();
-        let size = self.m.type_(cst.ty_idx()).byte_size().unwrap();
-        debug_assert_eq!(bytes.len(), size);
-        match size {
-            8 => {
-                let val = bytes.read_i64::<NativeEndian>().unwrap();
-                dynasm!(self.asm; mov Rq(reg.code()), QWORD val);
-            }
-            4 => {
-                let val = bytes.read_i32::<NativeEndian>().unwrap();
-                dynasm!(self.asm; mov Rq(reg.code()), DWORD val);
-            }
-            2 => {
-                let val = bytes.read_i16::<NativeEndian>().unwrap();
-                dynasm!(self.asm; mov Rw(reg.code()), WORD val);
-            }
-            1 => {
-                let val = bytes.read_i8().unwrap();
-                dynasm!(self.asm; mov Rq(reg.code()), val as i32);
-            }
-            _ => todo!("{}", size),
-        };
+        match self.m.const_(cidx) {
+            jit_ir::Const::I8(x) => dynasm!(self.asm; mov Rq(reg.code()), i32::from(*x)),
+            jit_ir::Const::I16(x) => dynasm!(self.asm; mov Rw(reg.code()), WORD *x),
```

Comment:
Complete bikeshed at this stage of development, but here I think it'd be better to use a 32-bit operand to avoid x86 [register stalls](https://fiigii.com/2020/02/16/Does-register-selection-matter-to-performance-on-x86-CPUs/#Partial-Register-Stall).

And the `DWORD` is redundant I think?

Tests seem to pass with:

```diff
diff --git a/ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs b/ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs
index 804f0909..faf5e087 100644
--- a/ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs
+++ b/ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs
@@ -828,8 +828,8 @@ impl<'a> X64CodeGen<'a> {
     fn load_const(&mut self, reg: Rq, cidx: jit_ir::ConstIdx) {
         match self.m.const_(cidx) {
             jit_ir::Const::I8(x) => dynasm!(self.asm; mov Rq(reg.code()), i32::from(*x)),
-            jit_ir::Const::I16(x) => dynasm!(self.asm; mov Rw(reg.code()), WORD *x),
-            jit_ir::Const::I32(x) => dynasm!(self.asm; mov Rq(reg.code()), DWORD *x),
+            jit_ir::Const::I16(x) => dynasm!(self.asm; mov Rd(reg.code()), i32::from(*x)),
+            jit_ir::Const::I32(x) => dynasm!(self.asm; mov Rq(reg.code()), *x),
             jit_ir::Const::I64(x) => dynasm!(self.asm; mov Rq(reg.code()), QWORD *x),
             jit_ir::Const::Ptr(_) => todo!(),
 ```

Reply:
I copied all this stuff over from the old code and have no idea about x86! It would probably be good if someone who actually knows x86 goes over this stuff in another PR.

Reply:
OK.